src/common/mfu_flist.c

```diff
@@ -1448,7 +1504,16 @@ static int map_spread(mfu_flist flist, uint64_t idx, int ranks, const void* args
     uint64_t global_idx = offset + idx;
 
     /* get global size of the list */
-    uint64_t global_size = mfu_flist_global_size(flist);
+    uint64_t global_size;
+
+    /* global size will be total obj ids instead of total files,
+     * if a non-posix copy is being performed */
+    bool* is_posix_copy = (bool*)args;
+    if (*is_posix_copy) {
+        global_size = mfu_flist_global_oid_size(flist);
+    } else {
+        global_size = mfu_flist_global_size(flist);
+    }
```

Comment:
@dsikich Minor: Just a thought...
What if `mfu_flist_global_size` did something like this?
```
uint64_t mfu_flist_global_size(mfu_flist bflist)
{
    flist_t* flist = (flist_t*) bflist;
#ifdef DAOS_SUPPORT
    uint64_t total_oids = flist->total_oids;
    if (total_oids > 0) {
        return total_oids;
    }
#endif
    uint64_t val = flist->total_files;
    return val;
}
```
The benefit would be that we wouldn't need `is_posix_copy` to be passed to `mfu_flist_spread`, which would simplify the function header.

Reply:
@dsikich , yes, I was also looking at whether we could avoid adding the ``is_posix_copy`` parameter to those functions.  Does this suggestion work instead?

Reply:
Taking a little closer look, two more things comes to mind.

After adding elements to an flist, we need to call ``mfu_flist_summarize``, which is a global collective.  All ranks must call this function, even if they didn't change the list.  This internally executes a bunch of collectives to do things like sum up the total number of items in the list (e.g., it calls allreduce and then sets ``total_files``).  You'd want to call this just after the ``daos_obj_list_oids`` function.

By the way, this feels like a "walk" but for DAOS objects.  We could wrap this in a new ``mfu_flist_walk_daos`` function, which would look something like:

```
int mfu_flist_walk_daos(daos_args, epoch, flist) {
  if (rank == 0) {
    daos_obj_list_oids(daos_args, &epoch, flist);
  }
  mfu_flist_summarize(flist);
}
```

Also, I think you could probably just rely on the existing ``total_files`` field rather than creating a separate ``total_oids`` field in the mfu_flist struct.  Although it's called "files", it really is a count of "items".  We may as well just overload it to be the oid count in this case.  That might simplify things quite a bit.

Reply:
@adammoody if I use the total_files field to store the number of objects I think it would be better to rename it to "total_items"? So that someone looking at it doesn't thing these are files.

Reply:
@adammoody there is a call to mfu_flist_summarize after listing the oids. I have daos_obj_list_oids, then I use a barrier to wait for rank 0 to retrieve all of the kids, then I call summarize, then mfu_flist_spread. Is this correct?

Reply:
@adammoody I think the barrier I put in is unnecessary before mfu_flist_summarize

Reply:
@adammoody I used @daltonbohning suggestion to remove the is_posix_copy variable from being passed to mfu_flist_spread by just checking if total_oids > 0.

Reply:
> 
> 
> @adammoody if I use the total_files field to store the number of objects I think it would be better to rename it to "total_items"? So that someone looking at it doesn't think these are files.

Yes, we can rename the field from total_files to total_items.  It's good by me if you want to do that now or in a separate PR.

Reply:
> 
> 
> @adammoody I think the barrier I put in is unnecessary before mfu_flist_summarize

That's right.  The mfu_flist_summarize includes synchronizing collectives, so it essentially has a barrier built in.